packages/server/modules/workspaces/graph/resolvers/workspaces.ts

```diff
@@ -437,37 +437,90 @@ export = FF_WORKSPACES_MODULE_ENABLED
       },
       WorkspaceMutations: {
         create: async (_parent, args, context) => {
-          const { name, description, logo, slug } = args.input
+          const {
+            name,
+            description,
+            logo,
+            slug,
+            enableDomainDiscoverabilityForDomain
+          } = args.input
+
+          const createWorkspace = commandFactory({
+            db,
+            eventBus,
+            operationFactory: ({ db, trx, emit }) => {
+              const createWorkspace = createWorkspaceFactory({
```

Comment:
Create + update + addDomain in a trx, which ensures that if any of those fail the workspace never gets committed

Reply:
wait, why not pass in the transaction as the db argument?

Reply:
@gjedlicska  what do you mean? that is what i'm doing here - passing in the trx as the db argument. Its only 1-2 lookup queries that use the main db connection, cause they don't need to be in a transaction

Reply:
i've made it only use the trx for all of them, if thats what u prefer